### PR TITLE
[WIRES] Sort observables in Tensor class

### PR DESCRIPTION
**Pull request by mariaschuld**
This PR temporarily solves the issue of how to order observables with the new wires management with a minimal solution.

Since the Tensor class will be redesigned soon, we limit ourselves here to implement a unique order by list comparison, which sorts lists lexicographically.

### CODE REVIEW

File: pennylane/operation.py [hunk 1]
```diff
@@ -1143,7 +1144,7 @@ def matrix(self):
         """
         # group the observables based on what wires they act on
         U_list = []
-        for _, g in itertools.groupby(self.obs, lambda x: x.wires.tolist()):
+        for _, g in itertools.groupby(self.obs, lambda x: x.wires.labels):
```
Comment (mariaschuld) [inline pennylane/operation.py:1147]: Same thing, just more elegant...

**Comment by mariaschuld** [comment]
@josh146, @co9olguy  and I were wondering if this ordering is what we want (for now)? 

It would compare the first elements of lists, and in case of a tie move on to the next one:

``print(sorted([['a'], ['a', 'c'], ['b', 'd'], ['b', 'a']]))``

**Comment by co9olguy** [comment]
Will defer to @josh146 for final approval here, since he understands the need for ordering in the Tensor class better than I do

**Comment by zeyueN** [comment]
Hey! Was very curious how the sorting is done so just peeked into this. Just wondering, should we worry about not having leading zeros for double digit int wires? eg. to prevent '10' appearing after '1' in this:

> ```python
> In [2]: sorted([str(i) for i in range(11)])
> Out[2]: ['0', '1', '10', '2', '3', '4', '5', '6', '7', '8', '9']
> ```

**Comment by mariaschuld** [comment]
> Hey! Was very curious how the sorting is done so just peeked into this. Just wondering, should we worry about not having leading zeros for double digit int wires? eg. to prevent '10' appearing after '1' in this:
> 
> > ```python
> > In [2]: sorted([str(i) for i in range(11)])
> > Out[2]: ['0', '1', '10', '2', '3', '4', '5', '6', '7', '8', '9']
> > ```

Good point, but then if I understand correctly, this wire ordering is just a means of internal bookkeeping and therefore to some extend arbitrary (as long as it is uniquely reproducable). In the long term we need a solution that does not rely on the fact that to us '10' comes after '2'.

**Pull request merged**